lib/types/accesslist/accesslist.go

```diff
@@ -81,73 +38,73 @@ type AccessList struct {
 	header.ResourceHeader
 
 	// Spec is the specification for the access list.
-	Spec AccessListSpec `json:"spec" yaml:"spec"`
+	Spec Spec `json:"spec" yaml:"spec"`
 }
 
-// AccessListSpec is the specification for an access list.
-type AccessListSpec struct {
+// Spec is the specification for an access list.
```

Comment:
What's the advantage of having almost duplicate structs here and generated structs from protobuf?

Reply:
There's a fairly substantial debate about it in https://github.com/gravitational/teleport/pull/28386, but overall there's value in decoupling our internal storage representation from our API where applicable. In this case, I think it's applicable, so I've made this separate resource.